### PR TITLE
EE-1085: add failing test for auction Error conversions, fix failures

### PR DESCRIPTION
**Pull request by henrytill**
Ref: https://casperlabs.atlassian.net/browse/EE-1085

When doing some related work, I discovered that `Error::InvalidValidatorSlotsValue` was missing conversion code, so here I'm introducing a regression test and fixing it.

### CODE REVIEW

File: types/src/system_contract_errors/auction.rs [hunk 1]
```diff
@@ -189,3 +193,25 @@ impl From<PurseLookupError> for Error {
...
+    #[test]
+    fn error_round_trips() {
+        for i in 0..=u8::max_value() {
+            match Error::try_from(i) {
+                Ok(error) if i <= MAX_ERROR_VALUE => assert_eq!(error as u8, i),
+                Ok(error) => panic!(
+                    "value of variant {} ({}) exceeds MAX_ERROR_VALUE ({})",
+                    error, i, MAX_ERROR_VALUE
+                ),
+                Err(_) if i > MAX_ERROR_VALUE => (),
```
Comment (xcthulhu) [inline types/src/system_contract_errors/auction.rs:212]: Maybe `Err(TryFromU8ForError(())) => assert!(i > MAX_ERROR_VALUE)` here?

Comment (henrytill) [reply types/src/system_contract_errors/auction.rs:215]: Yes I suppose that would force attention here if someone were to change the `Error` type in our `TryFrom` instance, which is probably a good thing.  Will do on both `Err` variants.

Comment (henrytill) [reply types/src/system_contract_errors/auction.rs]: Fixed.

**Review by EdHastingsCasperLabs — APPROVED** [review]

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**